### PR TITLE
Fix panic in cache informer

### PR DESCRIPTION
**Pull request by mfojtik**
*Description of changes:*

Not handling the tombstones in informer leads to panics:

```
E0902 18:40:00.201247       1 runtime.go:69] Observed a panic: &runtime.TypeAssertionError{_interface:(*runtime._type)(0x11545a0), concrete:(*runtime._type)(0x11c6960), asserted:(*runtime._type)(0x12c2160), missingMethod:""} (interface conversion: interface {} is cache.DeletedFinalStateUnknown, not *v1.ServiceAccount)
/go/src/github.com/aws/amazon-eks-pod-identity-webhook/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/go/src/github.com/aws/amazon-eks-pod-identity-webhook/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/go/src/github.com/aws/amazon-eks-pod-identity-webhook/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/lib/golang/src/runtime/panic.go:969
/usr/lib/golang/src/runtime/iface.go:260
/go/src/github.com/aws/amazon-eks-pod-identity-webhook/pkg/cache/cache.go:121
/go/src/github.com/aws/amazon-eks-pod-identity-webhook/vendor/k8s.io/client-go/tools/cache/controller.go:209
/go/src/github.com/aws/amazon-eks-pod-identity-webhook/vendor/k8s.io/client-go/tools/cache/controller.go:373
/go/src/github.com/aws/amazon-eks-pod-identity-webhook/vendor/k8s.io/client-go/tools/cache/delta_fifo.go:436
/go/src/github.com/aws/amazon-eks-pod-identity-webhook/vendor/k8s.io/client-go/tools/cache/controller.go:150
/go/src/github.com/aws/amazon-eks-pod-identity-webhook/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152
/go/src/github.com/aws/amazon-eks-pod-identity-webhook/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153
/go/src/github.com/aws/amazon-eks-pod-identity-webhook/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/go/src/github.com/aws/amazon-eks-pod-identity-webhook/vendor/k8s.io/client-go/tools/cache/controller.go:124
/usr/lib/golang/src/runtime/asm_amd64.s:1373
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/cache/cache.go [hunk 1]
```diff
@@ -118,7 +118,19 @@ func New(defaultAudience, prefix string, clientset kubernetes.Interface) Service
...
 			DeleteFunc: func(obj interface{}) {
-				sa := obj.(*v1.ServiceAccount)
+				sa, ok := obj.(*v1.ServiceAccount)
+				if !ok {
+					tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+					if !ok {
+						utilruntime.HandleError(fmt.Errorf("couldn't get object from tombstone %+v", obj))
+						return
+					}
+					pod, ok = tombstone.Obj.(*v1.ServiceAccount)
```
Comment (sjenning) [inline pkg/cache/cache.go:128]: should be `sa` not `pod` right?

**Review by sjenning — CHANGES_REQUESTED** [review]

File: pkg/cache/cache.go (hunk 1 above)
Comment (mfojtik) [reply pkg/cache/cache.go]: yep, fixed, goimports quirk also (missing import)

File: pkg/cache/cache.go [hunk 2]
```diff
@@ -118,7 +120,19 @@ func New(defaultAudience, prefix string, clientset kubernetes.Interface) Service
 				c.addSA(sa)
 			},
 			DeleteFunc: func(obj interface{}) {
-				sa := obj.(*v1.ServiceAccount)
+				sa, ok := obj.(*v1.ServiceAccount)
+				if !ok {
+					tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
+					if !ok {
+						utilruntime.HandleError(fmt.Errorf("couldn't get object from tombstone %+v", obj))
```
Comment (josselin-c) [inline pkg/cache/cache.go:127]: Can you use `klog.Errorf` to be consistent with what we use in the rest of the code base?
If we want amazon-eks-pod-identity-webhook to use `utilruntime.HandleError` this should happen in its own PR.

**Review by josselin-c — CHANGES_REQUESTED** [review]
Minor change, otherwise lgtm

File: pkg/cache/cache.go (hunk 2 above)
Comment (jqmichael) [reply pkg/cache/cache.go:127]: For reference, similar code in k8s/k8s.

https://github.com/kubernetes/kubernetes/blob/4ae11dac2e2cc8594c528a679db5861c0926b514/staging/src/k8s.io/sample-controller/controller.go#L357

**Review by jqmichael — APPROVED** [review]
Thanks for the fix! 

Merging given that correctness is more important than styling. Feel free to submit follow up PR to address styling comments.

**Comment by jqmichael** [comment]
Another comment was we could potentially get rid of the user cache given informer already cache everything in the Store/Indexer.

**Pull request merged**